vllm/config.py

```diff
@@ -277,6 +280,13 @@ def __init__(self,
         self.max_logprobs = max_logprobs
         self.disable_sliding_window = disable_sliding_window
         self.skip_tokenizer_init = skip_tokenizer_init
+        self.enable_sleep_mode = enable_sleep_mode
+
+        from vllm.platforms import current_platform
+
+        if self.enable_sleep_mode:
+            assert current_platform.is_cuda(), (
+                "Sleep mode is only supported on CUDA devices.")
```

Comment:
User facing errors should be exception (e.g., ValueError) instead of assert.

Reply:
fixed in [cbdbcea](https://github.com/vllm-project/vllm/pull/11743/commits/cbdbcea26c61315eff3127c98f3f1423c6a4c397)